lib/core/src/chain_swap.rs

```diff
@@ -380,6 +380,14 @@ impl ChainSwapHandler {
     async fn handle_amountless_update(&self, swap: &ChainSwap) -> Result<(), PaymentError> {
         let id = swap.id.clone();
 
+        // Since we optimistically persist the accepted receiver amount, if accepting a quote with
+        //  the swapper fails, we might still think it's accepted, so now we should get rid of the
+        //  old invalid accepted amount.
+        if swap.accepted_receiver_amount_sat.is_some() {
+            info!("Handling amountless update for swap {id} with existing accepted receiver amount. Erasing the accepted amount now...");
+            self.persister.update_accepted_receiver_amount(&id, None)?;
+        }
+
```

Comment:
Arn't we doing this anyway in `inspect_err`?

Reply:
We are, but even though it's unlikely, we can fail the update to None there. In those cases, this would eventually catch the inconsistency and correct it.